compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -483,19 +483,39 @@ static int64_t roundUpToPow2(int64_t size, bool predicate) {
   return llvm::PowerOf2Ceil(size);
 }
 
+/// Returns the nearest multiple of `vectorSize` of `size` if `predicate` is
+/// true. Otherwise, returns `size`.
+static int64_t roundUpToMultipleOfVectorSize(int64_t size, int64_t vectorSize,
+                                             bool predicate) {
+  if (!predicate) {
+    return size;
+  }
+  assert(size > 0 && "Negative size");
+  return llvm::alignTo(size, vectorSize);
+}
+
 /// Computes the maximum tile size that can be used to distribute a dimension
 /// based on its number of iterations and the native vector size used of the
 /// target. The resulting tile size will be a multiple of the provided vector
-/// size, except when `allowIncompleteTile` is set to true.
-static int64_t getMaxDistributionTileSize(int64_t lb, int64_t ub,
-                                          int64_t maxSize, int64_t vectorSize,
-                                          bool allowIncompleteTile = false) {
+/// size, except when `allowIncompleteTile` is set to true. If
+/// `enforceMultipleOfVectorSize` is set to true, the resulting tile size will
+/// be a multiple of `vectorSize`.
+static int64_t
+getMaxDistributionTileSize(int64_t lb, int64_t ub, int64_t maxSize,
+                           int64_t vectorSize, bool allowIncompleteTile = false,
+                           bool enforceMultipleOfVectorSize = false) {
```

Comment:
How much is the regression? I'm wondering if this is worth it as the distribution should eventually be based on the number of threads at runtime (what I've been working on). There's also the `allowIncompleteTile` option that is kind of doing something similar but slightly different. Would it be possible to unify these two options